### DISCUSSION TITLE
How to generate typescript type when using `await Assets.load(jsonUrl)`?

### DISCUSSION THREAD

**Discussion by SelfDevTV** [pixijs/pixijs#8848] [Q&A]
Hi guys,

new Pixi.js user here.
I created a new vanilla typescript project and started playing around with it .

One thing I can't figure out is, how to get the typescript type when I use `const sheet = await Assets.load(jsonUrl)`
When I use then `sheet.` I don't have any auto completion. 

What  I expect is that I can for one use `sheet.textures` and then it also autocompletes all the strings of the texture I defined in the loaded `.json`? (for example: `sheet.textures["player_1_idle"]` - "player_1_idle I expect to autocomplete).

The generated type of `const sheet` when  I hover is `any`.

This might be a completely dumb question but I can't figure this out.

Any ideas? Thx Geralt

- **`pixi.js` version**: _e.g. 7.0.4_
- **Browser & Version**: _e.g. Edge Latest_
- **OS & Version**: _e.g. Windows 11 64bit

**Comment by SuperSodaSea** [comment]
You need to specify the type by yourself with `await Assets.load<SomeType>(url)`, since the type cannot be dynamically inferred at runtime.

https://github.com/pixijs/pixijs/blob/d0c65eae42090f3be46d3575b600d190f00a6bdd/packages/assets/src/Assets.ts#L409-L412

**Reply by SelfDevTV** [reply]
Thank you very much. Can You please Show this to me with an Example? 

All i pass is a src to a spritesheet config Json generated by Texture Packer. How should my type look like then?

Thx

**Reply by bigtimebuddy** [reply]
```ts
import { Assets, Spritesheet } from 'pixi.js';

const sheet = await Assets.load<Spritesheet>('path/to/file.json');
```

**Reply by SelfDevTV** [reply]
@bigtimebuddy 
Tank you very much that works so now I get types like `.textures`. But how would an example look to also provide my type 
 and then get autocompletion on `sprite.textures["autocompletehere"]` - or is that not possible? I have so many sprites in this sheet and can't remember the exact names. 

Here is how I do this currently:

`const player = new Sprite(sheet.textures["player_move1.png"]);`

.. or ..
`const player = Sprite.from(sheet.textures["player_move1.png"]);`

Both give me the same typescript error (but the code works - later i add it to scene and it gets rendered):

>Element implicitly has an 'any' type because expression of type '"player_move1.png"' can't be used to index type 'Spritesheet | Dict<Texture<Resource>>'. Property 'player_move1.png' does not exist on type 'Spritesheet | Dict<Texture<Resource>>'.ts(7053)

So the only other possible way then is to manually go into the `.json` and look up the sprite name I want at this moment of time etc. right?

Thx so much for all your help.
